File: code/799最长连续不重复子序列.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
using namespace std;

const int N = 1e5 + 10;
int g[N], cnt[N];
int n;

int main() {
  cin >> n;
  for (int i = 0; i < n; i++)
    cin >> g[i];

  int res = 0;
  for (int i = 0, j = 0; i < n; i++) {
    cnt[g[i]]++;
    while (cnt[g[i]] > 1)
      cnt[g[j++]]--;
    res = max(i - j + 1, res);
  }

  cout << res << endl;

  return 0;
}

File: code/799最长连续不重复子序列.py
if __name__ == "__main__":
    n = int(input())
    g = list(map(int, input().split()))
    cnt = [0] * 100010

    j, res = 0, 0
    for i in range(n):
        cnt[g[i]] += 1
        while cnt[g[i]] > 1:
            cnt[g[j]] -= 1
            j += 1
        res = max(res, i - j + 1)
    print(res)
